src/main/java/core/basesyntax/service/impl/DataConverterServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverterService;
+import java.util.List;
+
+public class DataConverterServiceImpl implements DataConverterService {
+    private static final int TITTLE_POSITION = 1;
+    private static final int NECESSARY_COUNT_OF_ELEMENTS = 3;
+    private static final String COMMA_REGEX = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .skip(TITTLE_POSITION) //for skipping title
+                .map(this::convertStringToFruitTransaction)
+                .toList();
+    }
+
+    private FruitTransaction convertStringToFruitTransaction(String line) {
+        String[] fruitTransactionElements = line.split(COMMA_REGEX);
+        if (fruitTransactionElements.length != NECESSARY_COUNT_OF_ELEMENTS) {
```

Comment:
This check is good for ensuring the correct number of elements. However, consider also adding a check for the validity of the operation code and quantity format before proceeding to parse them. This can prevent `IllegalArgumentException` from `Operation.fromCode` or `NumberFormatException` from `Integer.parseInt` if the input is not as expected.